tools/mri_violations_resolver.php

```diff
@@ -0,0 +1,160 @@
+                mri_violations_log.LogID as join_id,
+                p.CenterID as Site,
+                violations_resolved.Resolved as Resolved
+            FROM mri_violations_log
+            LEFT JOIN mri_scan_type
+            ON (mri_scan_type.ID=mri_violations_log.Scan_type)
+            LEFT JOIN violations_resolved
+            ON (violations_resolved.ExtID=mri_violations_log.LogID
+            AND violations_resolved.TypeTable='mri_violations_log')
+            LEFT JOIN candidate c
+            ON (mri_violations_log.CandID=c.CandID)
+            LEFT JOIN session s
+            ON (mri_violations_log.CandID = s.CandID)
+            LEFT JOIN psc p
+            ON (p.CenterID = c.CenterID)
+            WHERE Resolved is NULL UNION SELECT PatientName,
+                TimeRun,
+                c.ProjectID as Project,
+                s.SubprojectID as Subproject,
+                MincFile,
+                null,
+                Reason,
+                SeriesUID,
+                md5(concat_WS(':',MincFile,PatientName,SeriesUID,TimeRun)) as hash,
+                MRICandidateErrors.ID as join_id,
+                p.CenterID as Site,
+                violations_resolved.Resolved as Resolved
+            FROM MRICandidateErrors
+            LEFT JOIN violations_resolved
+            ON (violations_resolved.ExtID=MRICandidateErrors.ID
+            AND violations_resolved.TypeTable='MRICandidateErrors')
+            LEFT JOIN candidate c
+            ON (SUBSTRING_INDEX(MRICandidateErrors.PatientName,'_',1)=c.PSCID)
+            LEFT JOIN session s
+            ON (c.CandID = s.CandID)
+            LEFT JOIN psc p
+            ON (p.CenterID = c.CenterID)
+            WHERE Resolved is NULL)
+            as v WHERE
+            v.problem = :pr and
+            v.series_description like :sd
+            ORDER BY v.TimeRun DESC";
+
+// Filter values to modify
+$where   = array(
+            'pr' => 'Protocol Violation',
+            'sd' => '%t1%',
+           );
+$results = $DB->pselect($query, $where);
+
+foreach ($results AS $result) {
+
+    $newlyResolved         = array();
+    $newlyResolved['hash'] = $result['hash'];
+    $newlyResolved['Resolved']   = 'inserted_flag';
+    $newlyResolved['User']       = 'lorisadmin';
+    $newlyResolved['ChangeDate'] = date("Y-m-d H:i:s");
+    $newlyResolved['TypeTable']  = 'mri_violations_log';
+    $newlyResolved['ExtID']      = $result['join_id'];
+
+    // Does a hash and a ExtID already exists?
+    $row = $DB->pselectRow(
+        "SELECT * FROM violations_resolved
+        WHERE hash = :ha and ExtID = :ex ",
+        array(
+         'ha' => $result['hash'],
+         'ex' => $result['join_id'],
+        )
+    );
+
+    // Insert if new
+    if (empty($row)) {
+        if ($confirm) {
+            $DB->insert('violations_resolved', $newlyResolved);
+            print "inserting\t";
+        }
+        print $result['hash'] . "\t" . $result['join_id'] . "\n";
```

Comment:
This should print newlyResolved -- to generate a report that users find meaningful. The hash isn't going to mean much to an average user running this script.